client/app/lib/components/table/MuiTableAdapter/MuiTableToolbar.tsx

```diff
@@ -19,14 +19,14 @@ const ToolbarContainer = ({ children }: ToolbarContainerProps): JSX.Element => (
 const MuiTableToolbar = (props: ToolbarProps): JSX.Element | null => {
   const { t } = useTranslation();
 
-  if (props.alternative?.when())
-    return <ToolbarContainer>{props.alternative.render()}</ToolbarContainer>;
-
-  if (props.renderNative)
-    return (
-      <ToolbarContainer>
-        <div className="flex w-full justify-between">
-          {props.onSearchKeywordChange && (
+  if (!props.alternative?.when() && !props.renderNative) return null;
+
+  return (
+    <ToolbarContainer>
+      <div className="flex w-full justify-between">
+        {(props.renderNative ||
+          (props.alternative?.when() && props.alternative.keepNative)) &&
+          props.onSearchKeywordChange && (
```

Comment:
what's `props.onSearchKeywordChange` for?

Reply:
it's been there since the beginning (even before this PR), and after I analysed it, this props is to ensure that `onSearchKeywordChange` is defined before the search bar is rendered (basically a gatekeeper)

Reply:
is there any case where was can choose not to define the `onSearchKeywordChange`? Otherwise this is dead code that should be removed.

Reply:
if you look into the file `client/app/lib/components/table/TanStackTableBuilder/useTanStackTableBuilder.tsx`, that value `onSearchKeywordChange` is actually always defined as `setSearchKeyword`, and that function is defined within that component itself through

```
const [searchKeyword, setSearchKeyword] = useState('')
```

so, to answer your question, that `onSearchKeywordChange` is always defined. I will remove that gatekeeper since it's not needed anyway